### ISSUE TITLE
Vue3 Support

### ISSUE THREAD

**Issue by dougalg** [RonaldJerez/vue-input-facade#44]
Hi, I am wondering if there are any plans for vue3 support for this package? Thanks a lot

**Comment by RonaldJerez** [comment]
Hello, the current project I am using this on doesn’t have any plans to upgrade to vue 3 yet. It’s not in my immediate schedule but eventually I will.

**Comment by andreasvirkus** [comment]
What about now? 👀

**Comment by kevinmelo** [comment]
What i did was changing the methods inside `directive.js` to the new ones for Vue3, until this lib supports Vue3

**Comment by RonaldJerez** [comment]
I honestly don't have a need for it yet, which is why I haven't prioritize it, but I did look into it briefly last year and stopped because styleguidest did not support vue 3 yet, and the documentations page requires it.  I created an issue to see if anyone wanted to contribute and help us move to storybook which does support vue 3.  

I'm more than open to contributions, even under beta if anyone wants to create a PR with the necessary changes for vue3.

**Comment by ankurk91** [comment]
Can you mention about vue 3 support in readme as well?

**Comment by iyerusad** [comment]
`"vue-input-facade": "version": "3.0.0-beta.2"`
`"vue": "version": "3.2.47"`

Vue3 seems like it wanted below, particularly filter was throwing errors. 
**Vue3 main.js**
```js
import { createApp } from 'vue';
const app = createApp(App);

import { InputFacade, facade, filter } from 'vue-input-facade';
app.component('InputFacade', InputFacade);
app.directive('facade', facade);
app.config.globalProperties.$filters = {
  facade: filter
};

**Comment by RonaldJerez** [comment]
Ah yes, filters are not supported in vue3. That needs to be removed.

**Comment by simonmaass** [comment]
When using v3 and nuxt 3.6.5 i get the following error:

Named export 'facade' not found. The requested module 'vue-input-facade' is a CommonJS module, which may not support all module.exports as named exports.

I created a pugin as follows:

```
import { facade } from 'vue-input-facade'

export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.vueApp.directive('facade', facade)
})

```

**Comment by z1m1n** [comment]
Any chance to have it released as stable?

**Comment by RonaldJerez** [comment]
> Any chance to have it released as stable?

Yes, it's been on my todo list, just need to update the documentation a bit.  I haven't seen any reported issues for the vue3 version, safe to assume its good to go stable.
